File: 03_currency_converter/python/solution.py
RATES = {
    "USD": 1.0,
    "EUR": 0.92,
    "INR": 83.0,
    "GBP": 0.79,
    "JPY": 144.0
}

def convert(amount, from_currency, to_currency):
    return (amount / RATES[from_currency]) * RATES[to_currency]

def main():
    print("Available currencies:", ", ".join(RATES.keys()))

    try:
        amount = float(input("Enter amount: "))
        from_currency = input("From currency: ").upper()
        to_currency = input("To currency: ").upper()

        if from_currency not in RATES or to_currency not in RATES:
            print("Unsupported currency.")
            return

        result = convert(amount, from_currency, to_currency)
        print(f"{amount:.2f} {from_currency} = {result:.2f} {to_currency}")

    except ValueError:
        print("Invalid amount entered.")

if __name__ == "__main__":
    main()

File: 03_currency_converter/java/Solution.java
import java.util.*;

public class Solution {

    private static final Map<String, Double> RATES = Map.of(
        "USD", 1.0,
        "EUR", 0.92,
        "INR", 83.0,
        "GBP", 0.79,
        "JPY", 144.0
    );

    private static double convert(double amount, String from, String to) {
        return (amount / RATES.get(from)) * RATES.get(to);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Available currencies: " + RATES.keySet());
        try {
            System.out.print("Enter amount: ");
            double amount = Double.parseDouble(sc.nextLine());

            System.out.print("From currency: ");
            String from = sc.nextLine().toUpperCase();

            System.out.print("To currency: ");
            String to = sc.nextLine().toUpperCase();

            if (!RATES.containsKey(from) || !RATES.containsKey(to)) {
                System.out.println("Unsupported currency.");
                return;
            }

            double result = convert(amount, from, to);
            System.out.printf("%.2f %s = %.2f %s%n", amount, from, result, to);

        } catch (NumberFormatException e) {
            System.out.println("Invalid amount entered.");
        }
    }
}

